Sources/xcproj/PBXProj.swift

```diff
@@ -3,8 +3,135 @@ import PathKit
+                lhs.fileElements == rhs.fileElements &&
+                lhs.configurationLists == rhs.configurationLists &&
+                lhs.buildConfigurations == rhs.buildConfigurations &&
+                lhs.variantGroups == rhs.variantGroups &&
+                lhs.targetDependencies == rhs.targetDependencies &&
+                lhs.sourcesBuildPhases == rhs.sourcesBuildPhases &&
+                lhs.shellScriptBuildPhases == rhs.shellScriptBuildPhases &&
+                lhs.resourcesBuildPhases == rhs.resourcesBuildPhases &&
+                lhs.frameworksBuildPhases == rhs.frameworksBuildPhases &&
+                lhs.headersBuildPhases == rhs.headersBuildPhases &&
+                lhs.nativeTargets == rhs.nativeTargets &&
+                lhs.fileReferences == rhs.fileReferences &&
+                lhs.projects == rhs.projects &&
+                lhs.versionGroups == rhs.versionGroups &&
+                lhs.referenceProxies == rhs.referenceProxies
+        }
+
+        // MARK: - Public Methods
+
+        public func addObject(_ object: PBXObject) {
+            switch object {
+            case let object as PBXBuildFile: buildFiles.append(object)
+            case let object as PBXAggregateTarget: aggregateTargets.append(object)
+            case let object as PBXContainerItemProxy: containerItemProxies.append(object)
+            case let object as PBXCopyFilesBuildPhase: copyFilesBuildPhases.append(object)
+            case let object as PBXGroup: groups.append(object)
+            case let object as PBXFileElement: fileElements.append(object)
+            case let object as XCConfigurationList: configurationLists.append(object)
+            case let object as XCBuildConfiguration: buildConfigurations.append(object)
+            case let object as PBXVariantGroup: variantGroups.append(object)
+            case let object as PBXTargetDependency: targetDependencies.append(object)
+            case let object as PBXSourcesBuildPhase: sourcesBuildPhases.append(object)
+            case let object as PBXShellScriptBuildPhase: shellScriptBuildPhases.append(object)
+            case let object as PBXResourcesBuildPhase: resourcesBuildPhases.append(object)
+            case let object as PBXFrameworksBuildPhase: frameworksBuildPhases.append(object)
+            case let object as PBXHeadersBuildPhase: headersBuildPhases.append(object)
+            case let object as PBXNativeTarget: nativeTargets.append(object)
+            case let object as PBXFileReference: fileReferences.append(object)
+            case let object as PBXProject: projects.append(object)
+            case let object as XCVersionGroup: versionGroups.append(object)
+            case let object as PBXReferenceProxy: referenceProxies.append(object)
+            default: fatalError("Unhandled PBXObject type for \(object), this is likely a bug / todo")
+            }
+        }
+
+        public func getReference<T: PBXObject>(_ reference: String) -> T? {
```

Comment:
This doesn't need to be generic. The other previous one was as the old temporary cache didn't have other explicit dictionaries to access. If someone knew which type they wanted they can now just use one of the explicit dictionaries. (which also doesn't have to loop through every collection)
This can just return a `PBXObject?`